### PR TITLE
wsdl2java toolchain article

### PR DESCRIPTION
**Pull request by madmas**
article about getting wsdl2java working

### CODE REVIEW

File: _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+featuredImage: gitlab-ci-pipelines
+excerpt: "Build a multiproject spanning Pipeline triggered by Tags in Gitlab CI"
+permalink: '/2024/05/28/cxf-jaxb-toolexception-wsdl2java.html'
+categories: [CXF, JAXB, wsdl2java, XML, SOAP, Code Generation, Gradle]
+header:
+  text: Apache CXF ToolException with JAXB / wsdl2java 
+  image: post
+---
+
+Imagine, in a project you are asked to consume a SOAP webservice. Yes, SOAP is aged as stone BUT it is also very percisely specified as a interface, which make it still a common choice when it comes to intergrate systems. And only integrated systems deliver great benefit on the one hand as well as (old) legacy system have proven their business case which is why they are worth to integrate. (There is *gold* in legacy.)
```
Comment (grigala) [inline _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md:14]: I'd actually fully reformulate it as it's a bit hard to follow, but in case you desire to keep it as is, here are some typo corrections.
```suggestion
Imagine, in a project you are asked to consume a SOAP webservice. Yes, SOAP is aged as stone BUT it is also very precisely specified as an interface, which make it still a common choice when it comes to integrate systems. And only integrated systems deliver great benefit on the one hand as well as (old) legacy system have proven their business case which is why they are worth to integrate. (There is *gold* in legacy.) 
```

File: _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+permalink: '/2024/05/28/cxf-jaxb-toolexception-wsdl2java.html'
+categories: [CXF, JAXB, wsdl2java, XML, SOAP, Code Generation, Gradle]
+header:
+  text: Apache CXF ToolException with JAXB / wsdl2java 
+  image: post
+---
+
+Imagine, in a project you are asked to consume a SOAP webservice. Yes, SOAP is aged as stone BUT it is also very percisely specified as a interface, which make it still a common choice when it comes to intergrate systems. And only integrated systems deliver great benefit on the one hand as well as (old) legacy system have proven their business case which is why they are worth to integrate. (There is *gold* in legacy.) 
+
+So while integrating that SOAP webservice, you try to generate some Java classes from the XML files provided and suddently, you see:
```
Comment (grigala) [inline _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md:16]:
```suggestion
So while integrating that SOAP webservice, you try to generate some Java classes from the XML files provided and suddenly, you see:
```

File: _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+task copyGeneratedFilesToSourceFolder(type: Copy) {
+	dependsOn 'wsdl2java'
+	from "${project.buildDir}/generated-sources/wsdl2java"
+	include "**/*.java"
+	into "${project.projectDir}/src/main/java/"
+}
+```
+
+But as i transferred this into the lager project, I got the error mentioned above:
```
Comment (grigala) [inline _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md:86]: 🍻 
```suggestion
But as i transferred this into the larger project, I got the error mentioned above:
```

File: _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+```
+org.apache.cxf.tools.common.ToolException: Tools plugin provider jaxb context init failed
+```
+
+To get more information, try running my build with the `--stacktrace`flag, but it does not give a more detailled output or more hints than the errormessage above.
```
Comment (grigala) [inline _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md:92]:
```suggestion
To get more information, tried running my build with the `--stacktrace` flag, but it did not give a more detailed output or more hints than the error message above.
```

File: _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+Going one step further with `--debug` , one finds himself flooded with output - but having a closer look and goind through it pay off. We find
```
Comment (grigala) [inline _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md:94]:
```suggestion
Going one step further with `--debug`, you find yourself flooded with an output - but having a closer look and going through it payed off. We find 
```

**Review by grigala — CHANGES_REQUESTED** [review]
Didn't have time to filly go through it and in more detail. 

But, I think it needs polishing to make it easy to read and follow.

**Comment by kress04** [comment]
lgtm, will merge it right away
